src/ucs/sys/topo.h

```diff
@@ -0,0 +1,80 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TOPO_H
+#define UCS_TOPO_H
+
+#include <ucs/type/status.h>
+#include <limits.h>
+#include <stdio.h>
+
+BEGIN_C_DECLS
+
+/** @file topo.h */
+
+typedef struct ucs_sys_bus_id {
+    unsigned int domain;   /* range: 0 to ffff */
+    unsigned int bus;      /* range: 0 to ff */
+    unsigned int slot;     /* range: 0 to 1f */
+    unsigned int function; /* range: 0 to 7 */
+} ucs_sys_bus_id_t;
+
+
+/**
+ * @ingroup UCS_RESOURCE
+ * System Device abstraction
+ */
+typedef struct ucs_sys_device {
+    unsigned int id;               /**< Index of the unit */
+    int numa_node;                 /**< NUMA node assoicated with the device*/
+    const ucs_sys_bus_id_t bus_id; /**< bus ID of of the device if applicable.
```

Comment:
why is this field const? how would the implementation initialize it? it's not c++ you don't have a constructor..

Reply:
removed const qualifier